File: segundo/1er cuatri/PDOO/Irrgarten_java/src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;
import java.util.ArrayList;
/**
 *
 * @author alvarogz
 */
public class FuzzyPlayer extends Player {
    public FuzzyPlayer(Player other){
        
        super(other.getNumber(), other.getIntelligence(), other.getStrength());
        setPos(other.getRow(), other.getCol());
    }
    
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        int size = validMoves.size();
        boolean contained = validMoves.contains(direction);
        Directions preference;
        
        if (size == 0 || !contained){
            preference = validMoves.get(0);
        }
        else{
            preference = direction;
        }
        
        return Dice.nextStep(preference, validMoves, super.getIntelligence());
    }
    
    @Override
    public float attack(){
    return Dice.intensity(super.getStrength()) + super.attack();

    //return Dice.intensity(super.getStrength()) + super.sumWeapons();
    }
    
    @Override
    protected float defensiveEnergy(){
        //        return Dice.intensity(super.getIntelligence())+ super.sumShields();
        return Dice.intensity(super.getIntelligence())+ super.defensiveEnergy();
    }
    
    @Override
    public String toString(){
        return "Fuzzy" + super.toString();
    }
}

File: segundo/1er cuatri/PDOO/irrgarten_ruby/Irrgarten/fuzzy_player.rb
# frozen_string_literal: true
require_relative 'player'
require_relative 'dice'

class FuzzyPlayer < Player
  def initialize(other)
    super(other.number, other.intelligence, other.strength)
    set_pos(other.row, other.col)
  end

  def move(direction, valid_moves)
    size = valid_moves.size
    contained = valid_moves.include?(direction)

    if size == 0 || !contained
      preference = valid_moves[0]
    else
      preference = direction
    end

    Dice.next_step(preference, valid_moves, @intelligence)
  end

  def attack
    Dice.intensity(@strength) + super.attack
  end

  def defensive_energy
    Dice.intensity(@intelligence ) + super.defensive_energy
  end

  def to_s
    "Fuzzy: #{super.to_s}"
  end
  protected :defensive_energy
end
